### ISSUE TITLE
Changing CSS in RCL forces app restart when using BlazorWebView

### ISSUE THREAD

**Issue by jburman** [dotnet/aspnetcore#42871]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Scenario: Running WinForm app with dotnet watch. App has all Blazor components and resources in a Razor Class Library.  

Hot reload works with component changes, but changes to .css files are detected as rude edits and forces the app to restart.

### Expected Behavior

Css changes should be hot-reloaded instead of restarting app.

### Steps To Reproduce

Create WinForms hybrid blazor app, reference an RCL with .css file and run from command line with dotnet watch.

### .NET Version

6.0.301

**Comment by javiercn** [comment]
Might be related to https://github.com/dotnet/sdk/issues/26762

**Comment by jburman** [comment]
Just to add... this was with a stand alone css file under the wwwroot folder inside the RCL.

**Comment by TanayParikh** [comment]
Related to https://github.com/dotnet/maui/issues/7479?

**Comment by javiercn** [comment]
@TanayParikh I am giving this to you since you have more context.

**Comment by TanayParikh** [comment]
@jburman can you please confirm if you're still encountering this issue in the latest VS and .NET 7 RC2? If so, please provide information on the exact VS and SDK versions you're using.

**Comment by jburman** [comment]
TLDR: no changes


Hi, the original issue I reported was with dotnet watch. I am still seeing the same behavior with .net 7 SDK installed, but I did some additional testing...

I just installed VS 17.4.0 Preview 4.0 and have SDK 7.0.100-rc.2.22477.23.  I also updated my test project to net7.0-windows.

- If I run the program with (or without) debugger attached, VS will not allow me to save changes to the CSS file and it thinks that the file is locked. What is strange is that I can edit the css file without any issues outside of VS and VS will hot-reload the CSS into the app.
- If I run the app with dotnet watch, VS also says that the file is locked and will not save it, but I can still edit the file outside of VS.  However, dotnet watch still thinks it's a rude edit and wants to restart the whole app. I primarily do all of my work through dotnet watch.

For what it's worth, I also have VS 17.3.6 Enterprise installed and I rolled the projects back to .net 6 and they behave the same as above.

**Comment by TanayParikh** [comment]
Thanks @jburman. Just to confirm, are you seeing https://github.com/dotnet/maui/issues/9197 by any chance?

I would've expected https://github.com/dotnet/maui/pull/9645 to resolve the Save As issue, and I believe 17.4 Preview 4 should have that. 

cc/ @BillHiebert as we've been discussing these issues.

**Comment by TanayParikh** [comment]
The file lock issues described above are likely a result of https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1659321.

> However, dotnet watch still thinks it's a rude edit and wants to restart the whole app.

@BillHiebert thoughts on this?

**Comment by TanayParikh** [comment]
Closing this out now as there have been many changes in this area and this issue should now be resolved. If you're still experiencing this issue on the latest VS Preview, with the latest SDK, please let us know by either commenting on this issue or creating a new issue should this issue be locked.
